pkg/gds/contacts/contacts.go

```diff
@@ -0,0 +1,75 @@
+package contacts
+
+import (
+	"errors"
+	"time"
+
+	"github.com/trisacrypto/directory/pkg/models/v1"
+)
+
+// Returns True if a Contact is not nil and has an email address.
+func ContactHasEmail(contact *models.Contact) bool {
+	return contact != nil && contact.Email != ""
+}
+
+// Returns True if a Contact is verified.
+func ContactIsVerified(contact *models.Contact) (verified bool) {
+	// Return zero-valued defaults with no error if extra is nil.
+	if contact == nil {
+		return false
+	}
+	return contact.Verified
+}
+
+// GetContactVerification token and verified status from the extra data field on the Contact.
+func GetContactVerification(contact *models.Contact) (string, bool) {
+	// Return zero-valued defaults with no error if extra is nil.
+	if contact == nil {
+		return "", false
+	}
+	return contact.Token, contact.Verified
+}
+
+// SetContactVerification token and verified status on the Contact record.
+func SetContactVerification(contact *models.Contact, token string, verified bool) (err error) {
+	if contact == nil {
+		return errors.New("cannot set verification on nil contact")
+	}
+
+	// Set contact verification.
+	contact.Verified = verified
+	contact.Token = token
+	return nil
+}
+
+// GetEmailLog from the extra data on the Contact record.
+func GetEmailLog(contact *models.Contact) (_ []*models.EmailLogEntry) {
+	// If the extra data is nil, return nil (no email log).
+	if contact == nil {
+		return nil
+	}
+	return contact.EmailLog
+}
+
+// Create and add a new entry to the EmailLog on the extra data on the Contact record.
+func AppendEmailLog(contact *models.Contact, reason, subject string) (err error) {
```

Comment:
Since this now only returns an error if the contact is nil, I'm wondering if we should just make this a method on `models.Contact`? The way I'm picturing us using this is:

```golang
// Retrieve contact from the database
var contact *models.Contact
if contact, err = db.RetrieveContact(email); err != nil {
    return err
}

...

// Append an entry to the log (in some method that sends an email)
contact.AppendEmailLog(reason, subject)
```

`contact` won't be nil because now we only get it from a database lookup, and we should be returning an error instead. Before we had to worry about if it was nil because it was a field on a VASP.

So as far as I can tell, we don't need to return an error from `AppendEmailLog` anymore, what do you think?